src/lando/pushlog/pushlog.py

```diff
@@ -0,0 +1,144 @@
+import logging
+from contextlib import contextmanager
+from typing import Optional
+
+from django.db import transaction
+
+from lando.main.models.repo import Repo
+from lando.main.scm.commit import Commit as SCMCommit
+from lando.pushlog.models.commit import Commit
+from lando.pushlog.models.push import Push
+
+logger = logging.getLogger(__name__)
+
+
+@contextmanager
+def PushLogForRepo(repo: Repo, user: str):
+    """
+    Context manager allowing to incrementally build push information, and only submit it
+    when complete.
+
+    WARNING: Do not use record_push() on the returned PushLog, as the context manager
+    will take care of it automatically. Calling it separately may lead to duplicates.
+    """
+    if repo.pushlog_disabled:
```

Comment:
Nit: this could be much flatter if we did and early `return` in the `pushlog_disabled` path. Also, we could run the `record_push` outside of the `else` clause since it runs on success regardless.